psydac/core/bsplines.py

```diff
@@ -600,5 +600,21 @@ def basis_ders_on_quad_grid( knots, degree, quad_grid, nders ):
             ders = basis_funs_all_ders( knots, degree, xq, span, nders )
             basis[ie,:,:,iq] = ders.transpose()
 
+    if normalize:
+        x = scaling_matrix(degree, ne+degree, knots)
+        basis *= x[0]
+
     return basis
 
+#==============================================================================
+def scaling_matrix(p, n, T):
+    """Returns the scaling array for M-splines.
+    It is an array whose elements are (p+1)/(T[i+p+1]-T[i])
+
```

Comment:
We should add to this comment how the `scaling_matrix` is used to convert B-splines into M-splines and vice versa. Moreover, I do not see why we should call it "matrix", given that it is just a list of scalar values.